src/main/java/org/fenixedu/academic/ui/struts/action/student/ShowStudentPortalDA.java

```diff
@@ -62,24 +63,26 @@
     @Override
     public ActionForward execute(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response)
             throws Exception {
-        List<StudentPortalBean> studentPortalBeans = new ArrayList<StudentPortalBean>();
-        List<String> genericDegreeWarnings = new ArrayList<String>();
+        final List<StudentPortalBean> studentPortalBeans = new ArrayList<StudentPortalBean>();
+        final List<String> genericDegreeWarnings = new ArrayList<String>();
 
         final Student student = AccessControl.getPerson().getStudent();
         if (student != null) {
-            for (Registration registration : student.getActiveRegistrationsIn(ExecutionSemester.readActualExecutionSemester())) {
-                DegreeCurricularPlan degreeCurricularPlan = registration.getActiveDegreeCurricularPlan();
-                if (registration.getAttendingExecutionCoursesForCurrentExecutionPeriod().isEmpty() == false) {
-                    studentPortalBeans.add(new StudentPortalBean(registration.getDegree(), student, registration
-                            .getAttendingExecutionCoursesForCurrentExecutionPeriod(), degreeCurricularPlan));
-                }
-                if (hasSpecialSeasonEnrolments(student)) {
-                    genericDegreeWarnings.addAll(getEnrolmentPeriodCoursesAfterSpecialSeason(degreeCurricularPlan));
-                } else {
-                    genericDegreeWarnings.addAll(getEnrolmentPeriodCourses(degreeCurricularPlan));
+            for (final Registration registration : student.getAllRegistrations()) {
+                if (considerRegistration(registration.getLastStateType())) {
```

Comment:
This filter is not necessary, the way to validate "acceptable" registrations for this is way to specific and is subject to get outdated with more states being added eventually.
The check is done in each registration for attends.